noise.go

```diff
@@ -58,3 +62,18 @@ func (s *NebulaCipherState) DecryptDanger(out, ad, ciphertext []byte, n uint64,
 		return []byte{}, nil
 	}
 }
+
+func sha256KdfFromString(secret string) ([]byte, error) {
+	if len(secret) < 8 {
+		err := ("PSK too short!")
+		return nil, fmt.Errorf("%s", err)
+	}
+	hmacKey := make([]byte, sha256.Size)
+	hash := sha256.New
+	hkdfer := hkdf.New(hash, []byte(secret), nil, nil)
+	n, err := io.ReadFull(hkdfer, hmacKey)
+	if n != len(hmacKey) || err != nil {
+		return nil, fmt.Errorf("%s", err)
```

Comment:
fmt.Errorf does nothing but toss away the original type of the error, why not just `return nil, err` ? Or provide another message and wrap it with `%w` (see "Wrapping errors with %w" in https://blog.golang.org/go1.13-errors)